src/tracer.js

```diff
@@ -253,6 +273,38 @@ export default class Tracer {
     }
 
     /**
+     * Reapplies sampling if the parent span isn't sampled
+     *
+     * @param parent the parent span context
+     * @param ctx the span context
+     * @param operationName the operation name for the child
+     * @param tags tags to be applied by the sampler
+     */
+    handleUpsampling(parent: SpanContext, ctx: SpanContext, operationName: string, tags: any) {
+        if (!this._upsampling.enabled) {
+            return
+        }
+
+        // Run upsampler on every span
+        if (this._upsampling.allowOnInnerSpans) {
+            if (this._sampler.isSampled(operationName, tags)) {
```

Comment:
I did not ask for it. Your earlier version was applying upsampling to _all spans_, and I said if you want to do that, it has to be opt-in, with default being only first-in-process span.